docs/current_docs/cookbook/snippets/clone-git-repository-ssh/typescript/index.ts

```diff
@@ -0,0 +1,23 @@
+import { dag, Directory, Container, object, func, Socket } from "@dagger.io/dagger"
+
+@object()
+class MyModule {
+  /**
+    Demonstrates an SSH-based clone requiring a user-supplied sshAuthSocket.
+    
+    For the reasoning behind explicit socket forwarding, see:
+    /path/to/security-by-design
+    You can also avoid passing a socket if you prefer the Directory pattern,
+    e.g. dagger call someFunc --dir git@github.com:org/repo@main
+   */
+  @func()
+  cloneWithSsh(repository: string, ref: string, sock: Socket): Container {
+    const repoDir = dag.git(repository, { sshAuthSocket: sock }).ref(ref).tree()
+
+    return dag
+      .container()
+      .from("alpine:latest")
+      .withDirectory("/src", repoDir)
+      .withWorkdir("/src")
+  }
+}
```

Comment:
Rather than creating a new example where the SSH sock is passed through explicitly as an argument.

Why not update the existing cookbook example to take a `Directory` for `repository` instead of a `string`?

That should demonstrate that from the Git repository URL, Dagger can clone the Git repository over HTTPS or SSH (provided SSH is configured)?